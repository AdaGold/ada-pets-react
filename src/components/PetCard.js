import React from 'react';
import PropTypes from 'prop-types';

import 'bootstrap/dist/css/bootstrap.min.css';
import './PetCard.css';

import speciesEmoji from '../speciesEmoji';


const PetCard = ({ id, name, species, about, location, deletePetCallback, selectPetCallback }) => {
  return (
    <div className="card pet-card">

      <section className="pet-card--header">
        {speciesEmoji(species)} {id} - {name}
        <div>
          <button
            onClick={() => { selectPetCallback(id) }}
            className="btn btn-primary pet-card--select-pet-btn"
          >
            Select
        </button>
          <button
            type="button"
            className="btn btn-danger pet-card--close-btn"
            aria-label="Close"
            onClick={() => { deletePetCallback(id) }}
          >
            Close
        </button>
        </div>
      </section>
      <section className="pet-card--body">
        {about.length > 128 ? `${ about.substring(0, 128) }...` : about}
      </section>
      <section className="pet-card--footer text-muted">
        {location}
      </section>
    </div>
  );
};

PetCard.propTypes = {
  id: PropTypes.number.isRequired,

  name: PropTypes.string.isRequired, 
  species: PropTypes.string.isRequired, 
  about: PropTypes.string.isRequired, 
  location: PropTypes.string,
  deletePetCallback: PropTypes.func.isRequired,
  selectPetCallback: PropTypes.func.isRequired,
}

export default PetCard;