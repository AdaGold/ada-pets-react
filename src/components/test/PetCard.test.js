import React from 'react';
import { render, cleanup } from '@testing-library/react'
import PetCard from '../PetCard';

describe('PetCard', () => {
  afterEach(cleanup);

  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(<PetCard
      id={1}
      name={"Samson"}
      species={"Cat"}
      about={"A very awesome cat!  Don't touch the hair!"}
      location={"Seattle, WA"}
      deletePetCallback={() => { }}
      selectPetCallback={() => { }}
    />);

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  test('The "selectPetCallback" function is called when the `select` button is clicked on', () => {

    // Arrange
    // Create a mock callback function
    const selectPet = jest.fn();

    // Render a PetCard
    const container = render(<PetCard
      id={1}
      name={"Samson"}
      species={"Cat"}
      about={"A very awesome cat!  Dont' touch the hair"}
      location={"Seattle, WA"}
      deletePetCallback={() => { }}
      selectPetCallback={selectPet}
    />);

    // Act
    // Find the "Select" button
    const selectButton = container.getByText(/Select/);
    // Trigger a 'click' event
    selectButton.click();

    // Assert
    // Verify that the callback function was called.
    expect(selectPet).toHaveBeenCalled();
  });
});