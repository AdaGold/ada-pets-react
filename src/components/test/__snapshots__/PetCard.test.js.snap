// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetCard that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div
    class="card pet-card"
  >
    <section
      class="pet-card--header"
    >
      ‚ùì 1 - Samson
      <button
        class="btn btn-primary pet-card--select-pet-btn"
      >
        Select
      </button>
      <button
        aria-label="Close"
        class="btn btn-danger pet-card--close-btn"
        type="button"
      >
        Close
      </button>
    </section>
    <section
      class="pet-card--body"
    >
      A very awesome cat!  Don't touch the hair!
    </section>
    <section
      class="pet-card--footer text-muted"
    >
      Seattle, WA
    </section>
  </div>
</DocumentFragment>
`;
